### ISSUE TITLE
KubeAPI Server can't get accessed

### ISSUE THREAD

**Issue by jia-jerry** [gardener/gardener-extension-provider-alicloud#25]
**What happened**:
After set ExternalTrafficPolicy of kube-apiserver service to local, some shoot clusters can't be accessed. The node port exposed on the node where apiserver pod is running can't be connected. After investigated by TCPDump,  the *syn* tcp packets from Cloud SLB probers are not responded. It is weird. It could be iptable rules issue because some shoots are normal. Need some time to track. But as it is a very critical issue for Gardener users, let's set *ExternalTrafficPolicy* back to *cluster* for now.
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Not always. All malfunctional shoots are in Hangzhou region while seed is in Shanghai.
**Anything else we need to know?**:

**Environment**:

- Gardener version (if relevant):
- Extension version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by jia-jerry** [comment]
I re-opened issue and keep work on it

**Comment by jia-jerry** [comment]
#26 temporarily solved the API access issue. But we still need to set kube-apiserver service's ExternalTrafficPolicy to Local as possible, because it will save the unnecessary network traffic.

**Comment by rfranzke** [comment]
Yes. I'm asking why it's suddenly okay again now to set the policy to `Local`. Earlier, you reverted it back to `Cluster` because there were problems - what were these problems and why/how are they solved now so that it's save to set it back to `Local`?

**Comment by jia-jerry** [comment]
> Yes. I'm asking why it's suddenly okay again now to set the policy to `Local`. Earlier, you reverted it back to `Cluster` because there were problems - what were these problems and why/how are they solved now so that it's save to set it back to `Local`?

I was planing to answer your question in PR #135 while I was preparing fix of Format issue :) .

One possible reason I could guess. At that time, the shoot network CIDRs for service/pod IP are 100.XXX. This is a carrier level CIDR and it is used in AliCloud internally. In Hangzhou, there could be more internal services using 100.XXX CIDR because AliCloud grew in Hangzhou. So it is more likely for shoots to conflict with AliCLoud internal services(SLB prober's source IPs are also in 100.XXX). And also, kube-api server happens to have VPN to shoot network... Based on previous 2 reasons, we could explain why 1. Only KubeAPIServer service had issue while Nginx/VPN in Shoot worked. 2. Why Shanghai worked while Hangzhou failed.

Now, all shoot networks have been fixed with private network CIDRS(like 172.16.XXX). I used fix in PR #135 to test in ac-canary landscape for 1 week. No issue found.
